vector/src/main/java/im/vector/activity/MediasPreviewerActivity.java

```diff
@@ -0,0 +1,53 @@
+package im.vector.activity;
+
+import android.os.Bundle;
+import android.app.Activity;
+import android.support.design.widget.FloatingActionButton;
+import android.view.View;
+import android.widget.ImageView;
+
+import org.matrix.androidsdk.data.RoomMediaMessage;
+import org.matrix.androidsdk.util.Log;
+
+import java.util.ArrayList;
+
+import im.vector.R;
+
+import static im.vector.activity.JoinScreenActivity.LOG_TAG;
+
+public class MediasPreviewerActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (CommonActivityUtils.shouldRestartApp(this)) {
+            Log.d(LOG_TAG, "onCreate : restart the application");
+            CommonActivityUtils.restartApp(this);
+            return;
+        }
+
+        if (CommonActivityUtils.isGoingToSplash(this)) {
+            Log.d(LOG_TAG, "onCreate : Going to splash screen");
+            return;
+        }
+
+        setContentView(R.layout.activity_medias_previewer);
+
+        ArrayList<RoomMediaMessage> sharedDataItems = new ArrayList<>(RoomMediaMessage.listRoomMediaMessages(getIntent(), RoomMediaMessage.class.getClassLoader()));
+        if (!sharedDataItems.isEmpty()) {
+            ImageView imagePreview = (ImageView) findViewById(R.id.image_previewer);
+            imagePreview.setImageURI(sharedDataItems.get(0).getUri());
+        }
```

Comment:
Only the first item ?
A webview might be more simple to manage animated gifs, videos or files